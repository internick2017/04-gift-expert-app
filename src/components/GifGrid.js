import React, { useState, useEffect } from 'react'

export const GifGrid = ({ category }) => {

    useEffect(() => {
        getGifs();
    }, [])
    const getGifs = async () => {

        const url = 'https://api.giphy.com/v1/gifs/search?q=Bleach&limit=10&api_key=dOv0LM1e9OjbMAvId5PDvdTySsgzD3ri';
        const resp = await fetch(url);
        const { data } = await resp.json();
        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })

        console.log(gifs)
    }

    return (
        <>
            <h3>{category}</h3>
            <button></button>
        </>
    )
}
