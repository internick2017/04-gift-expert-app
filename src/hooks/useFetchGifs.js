import { useEffect, useState } from "react"
import { getGifs } from '../helpers/getGifs';


export const useFetchGifs = (category) => {

    const [state, setstate] = useState({
        data: [],
        loading: true
    })

    useEffect(() => { // no pueden ser asycn

        getGifs(category)

            .then(img => {
                setTimeout(() => {
                    console.log(img)
                    setstate({
                        data: img,
                        loading: false
                    })
                }, 3000)
            })

    }, [category])



    return state; // {data:[], loading: true}
}