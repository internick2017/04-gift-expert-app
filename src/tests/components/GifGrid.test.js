import React from "react";
import { shallow } from 'enzyme';
import { useFetchGifs } from "../../hooks/useFetchGifs";
import { GifGrid } from "../../components/GifGrid";
jest.mock('../../hooks/useFetchGifs');

describe('pruebas en el GifGrid', () => {

    const category = 'Bleach';

    test('debe mostrarse correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot()

    });

    test('debe de mostrar items cuando se cargan imagenes', () => {

        const gifs = [
            {
                id: 'Nick',
                url: 'https://localhos/esto/aquello.com',
                title: 'Prueba de locos'
            }
        ]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
        

    });



});
