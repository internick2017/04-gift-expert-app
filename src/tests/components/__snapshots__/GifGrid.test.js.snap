// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el GifGrid debe de mostrar items cuando se cargan imagenes 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    Bleach
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="Nick"
      key="Nick"
      title="Prueba de locos"
      url="https://localhos/esto/aquello.com"
    />
    <GifGridItem
      id="2"
      key="2"
      title="Prueba de locos"
      url="https://localhos/esto/aquello.com"
    />
  </div>
</Fragment>
`;

exports[`pruebas en el GifGrid debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    Bleach
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
